sbol/identified.py

```diff
@@ -189,3 +189,251 @@ def update_uri(self):
+                        {
+                            replacement = getHomespace();
+                        }
+                        property_val.erase(pos, replace_target.size());
+                        property_val.insert(pos, replacement);
+                        property_store_copy[i_property_val] = property_val;
+                    }
+                }
+            }
+        }
+        new_obj.properties[store_uri] = property_store_copy;
+    }
+
+    // Set version
+
+    if (version != "")
+        new_obj.version.set(version);
+    else if (version == "" && ns == "")
+    {
+        if (this->version.size() > 0)
+            if (this->doc != NULL && this->doc == target_doc)  // In order to create a copy of the object in this Document, it's version must be incremented
+                new_obj.version.incrementMajor();
+            else
+            {
+                new_obj.version.set(this->version.get());  // Copy this object's version if the user doesn't specify a new one
+            }
+    }
+    else if (version == "" && ns != "")
+    {
+        if (this->version.size() > 0)
+            {
+                new_obj.version.set(this->version.get());  // Copy this object's version if the user doesn't specify a new one
+            }
+    }
+
+    string id;
+    if (Config::getOption("sbol_compliant_uris") == "True" && this->version.size() > 0)
+        id = new_obj.persistentIdentity.get() + "/" + new_obj.version.get();
+    else
+        id = new_obj.persistentIdentity.get();
+    new_obj.identity.set(id);
+
+    // Copy wasDerivedFrom
+    if (this->identity.get() != new_obj.identity.get())
+        new_obj.wasDerivedFrom.set(this->identity.get());  // When generating a new object from an old, point back to the copied object
+    else if (this->wasDerivedFrom.size() > 0)
+        new_obj.wasDerivedFrom.set(this->wasDerivedFrom.get()); // When cloning an object, don't overwrite the wasDerivedFrom
+
+    for (auto i_store = owned_objects.begin(); i_store != owned_objects.end(); ++i_store)
+    {
+        string store_uri = i_store->first;
+        if (target_doc && std::find(hidden_properties.begin(), hidden_properties.end(), store_uri) != hidden_properties.end() )
+            continue;
+
+        vector < SBOLObject* >& object_store = i_store->second;
+        for (auto i_obj = object_store.begin(); i_obj != object_store.end(); ++i_obj)
+        {
+            Identified& child_obj = (Identified&)**i_obj;
+
+            // Recurse into child objects and copy.  This should be after all other object properties are set, to ensure proper generation of new URIs with updated namespace and version
+            Identified& child_obj_copy = child_obj.copy(target_doc, ns, version);
+            new_obj.owned_objects[store_uri].push_back((SBOLObject*)&child_obj_copy);  // Copy child object
+            child_obj_copy.parent = &new_obj;
+            child_obj_copy.update_uri();
+        }
+    }
+    return new_obj;
+};
+'''
```

Comment:
What on earth is this?

If this is essentially commented out code, please delete it. Or maybe it's because it's a work in progress. But still, delete it. Git history can get you back to it.

Reply:
Deleted